Log debug message for InsufficientStorage exception

sharidas opened the pull request:
Catch the insufficient storage exception and log as
debug message instead of throwing the trace.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.
-->

## Description
<!--- Describe your changes in detail -->
Catch InsufficientStorage exception and log it as debug message.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://github.com/owncloud/core/issues/31284

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Catch InsufficientStorage exception and log it as debug message.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- Inorder to test this tried to comment out the lines https://github.com/owncloud/core/blob/master/apps/files/js/file-upload.js#L1026-L1033
and then reloaded the web page.
- Upload a file and found this message in the log file: ```{"reqId":"Yi4GPI1j7rbrrO76v0LX","level":0,"time":"2018-07-03T08:03:32+00:00","remoteAddr":"127.0.0.1","user":"user1","app":"no app in context","method":"PUT","url":"\/testing3\/remote.php\/dav\/files\/user1\/a.txt","message":"Can not create file a.txt due to insufficient storage"}```

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

patrickjahns commented:
How about using the [ExceptionLoggerPlugin](https://github.com/owncloud/core/blob/master/apps/dav/lib/Connector/Sabre/ExceptionLoggerPlugin.php)

That has a way of translating levels from exceptions to debug messages?

sharidas commented:
> How about using the ExceptionLoggerPlugin

That's where I am making the change, by removing the change which I have made in this PR. I got the hint because of failure that was happening in acceptance test. Thanks.

sharidas committed:
Log debug message for InsufficientStorage exception

Log a debug message, instead of writing detailed exception
message to the log file.

Signed-off-by: Sujith H <sharidasan@owncloud.com>

patrickjahns commented:
Please have a look at stable10 - https://github.com/owncloud/core/blob/stable10/apps/dav/lib/Connector/Sabre/ExceptionLoggerPlugin.php

It's still missing a forward port as it was blocked by the exception logging events

patrickjahns commented on apps/dav/lib/Connector/Sabre/ExceptionLoggerPlugin.php:
```diff
@@ -92,6 +92,12 @@ public function logException(\Exception $ex) {
 			return null;
 		}
+		//If the exception is InsufficientStorage, then log a debug message and return null
+		if (($ex->getPrevious() === null) && ($ex instanceof Exception\InsufficientStorage)) {
```
Why this? `$this->nonFatalExceptions` contains nonFatal exceptions which will be logged in debug mode

sharidas replied:
@patrickjahns  Oops, I guess I took the wrong path. I was trying to suppress the exception message too. My bad. I have updated the code.

patrickjahns replied:
Might have misunderstood me there - maybe your approach was the right one - was trying to understand why not using already existing functionality

sharidas replied:
@patrickjahns In this https://github.com/owncloud/core/issues/31284#issue-317988198 , `This should be probably treated like a normal log message`, was the reason I thought to suppress the message. But in fact you are right, considering this https://github.com/owncloud/core/issues/31284#issuecomment-399905245. Treating the exception message as debug one. The current change does what is required ;)

PVince81 reviewed (approved):
:+1: please backport

sharidas commented:
Backport to stable10 -> https://github.com/owncloud/core/pull/31978
